src/git/git.go

```
@@ -181,3 +184,122 @@ func CreatedRepo(r *http.Request) (*Repository, int, error) {
+	return "Bearer " + githubActionToken == headerValue
+}
+
+// CIHandler handles the request to add CI Files to a repo
+func CIHandler(w http.ResponseWriter, r *http.Request) {
+
+	if !validateActionRequest(r) {
+		log.Info("Request verification from Github Action: FAILED")
+		w.WriteHeader(403)
+		return
+	}
+	log.Info("Request verification from Github Action: SUCCESS")
+
+	if r.Method != "POST" {
+		w.WriteHeader(405)
+		log.Errorf("received a %s request, expected POST for CI Handler", r.Method)
+		return
+	}
+
+	r_body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(500)
+		log.Errorf("cannot read request body - %v", err)
+		return
+	}
+
+	var ci CIAction
+	if err := json.Unmarshal(r_body, &ci); err != nil {
+		w.WriteHeader(500)
+		log.Errorf("cannot unmarshal request body - %v", err)
+		return
+	}
+
+	var ciList []string
+	// add Default checks
+	ciList = append(ciList, "Default")
+	v := reflect.ValueOf(ci)
+	for i := 0; i < v.NumField(); i++ {
+		if v.Field(i).String() == "true" {
+			ciList = append(ciList, v.Type().Field(i).Name)
+		}
+	}
+	addChecks(ciList)
+	log.Info("Added checks to pre-commit-config.yaml")
+
+	// call a script to add this file to the repo
+
+	// get ssh_url of the repo
+	repositories, err := Repos()
+	if err != nil {
+		log.Error("Error getting repositories %v", err)
+		return
+	}
+	var repo Repository
+	for _, repository := range repositories {
+		if repository.Name == ci.Repo {
+			repo = repository
+		}
+	}
+	if repo.Name == "" {
+		log.Error("Repository not found")
+		return
+	}
+
+	for _, branch := range repo.Branches {
+		go addCI(repo.URL, repo.Name, branch)
+	}
+
+}
+
+func addChecks(ciList []string) {
+	// combine the data of all files in ciList to a file named .pre-commit-config.yaml
+
+	// create the file even if it exists
+	os.Create(PRE_COMMIT_CONFIG)
+	f, err := os.OpenFile(PRE_COMMIT_CONFIG, os.O_APPEND|os.O_WRONLY, 0600)
+	if err != nil {
+		log.Error("Error opening file:pre-commit.config")
+		return
```

Maybe return the err from here